otelconf/config_json.go

```diff
@@ -8,6 +8,126 @@ import (
 	"errors"
 )
 
+// UnmarshalJSON implements json.Unmarshaler.
+func (j *PushMetricExporter) UnmarshalJSON(b []byte) error {
+	var raw map[string]any
+	if err := json.Unmarshal(b, &raw); err != nil {
+		return err
+	}
+	type Plain PushMetricExporter
+	var plain Plain
+	if err := json.Unmarshal(b, &plain); err != nil {
+		return err
+	}
+	// console can be nil, must check and set here
+	if _, ok := raw["console"]; ok {
+		plain.Console = ConsoleExporter{}
+	}
+	*j = PushMetricExporter(plain)
+	return nil
+}
+
+// UnmarshalJSON implements json.Unmarshaler.
+func (j *SpanExporter) UnmarshalJSON(b []byte) error {
+	var raw map[string]any
+	if err := json.Unmarshal(b, &raw); err != nil {
+		return err
+	}
+	type Plain SpanExporter
+	var plain Plain
+	if err := json.Unmarshal(b, &plain); err != nil {
+		return err
+	}
+	// console can be nil, must check and set here
+	if _, ok := raw["console"]; ok {
+		plain.Console = ConsoleExporter{}
+	}
+	*j = SpanExporter(plain)
+	return nil
+}
+
+// UnmarshalJSON implements json.Unmarshaler.
+func (j *LogRecordExporter) UnmarshalJSON(b []byte) error {
+	var raw map[string]any
+	if err := json.Unmarshal(b, &raw); err != nil {
+		return err
+	}
+	type Plain LogRecordExporter
+	var plain Plain
+	if err := json.Unmarshal(b, &plain); err != nil {
+		return err
+	}
+	// console can be nil, must check and set here
+	if _, ok := raw["console"]; ok {
+		plain.Console = ConsoleExporter{}
+	}
+	*j = LogRecordExporter(plain)
+	return nil
+}
+
+// UnmarshalJSON implements json.Unmarshaler.
+func (j *BatchLogRecordProcessor) UnmarshalJSON(b []byte) error {
+	var raw map[string]any
+	if err := json.Unmarshal(b, &raw); err != nil {
+		return errors.Join(errUnmarshalingBatchLogRecordProcessor, err)
+	}
+	if _, ok := raw["exporter"]; raw != nil && !ok {
+		return newErrRequiredExporter("BatchLogRecordProcessor")
+	}
+	type Plain BatchLogRecordProcessor
+	var plain Plain
+	if err := json.Unmarshal(b, &plain); err != nil {
+		return errors.Join(errUnmarshalingBatchLogRecordProcessor, err)
+	}
+	if err := validateBatchLogRecordProcessor((*BatchLogRecordProcessor)(&plain)); err != nil {
+		return err
+	}
+	*j = BatchLogRecordProcessor(plain)
+	return nil
```

Comment:
Similar here, the double `Unmarshal` can be avoided.

```suggestion
	type Plain BatchLogRecordProcessor
	type shadow struct {
		Plain
		Exporter json.RawMessage `json:"exporter"`
	}
	var sh shadow
	if err := json.Unmarshal(b, &sh); err != nil {
		return errors.Join(errUnmarshalingBatchLogRecordProcessor, err)
	}
	if sh.Exporter == nil {
		return newErrRequiredExporter("BatchLogRecordProcessor")
	}
	// Hydrate the exporter into the underlying field.
	if err := json.Unmarshal(sh.Exporter, &sh.Plain.Exporter); err != nil {
		return errors.Join(errUnmarshalingBatchLogRecordProcessor, err)
	}
	err := validateBatchLogRecordProcessor((*BatchLogRecordProcessor)(&sh.Plain))
	if err != nil {
		return err
	}
	*j = BatchLogRecordProcessor(sh.Plain)
	return nil
```

Similar for `BatchSpanProcessor` and `PeriodicMetricReader`.